app/modules/asset.rb

```diff
@@ -0,0 +1,148 @@
+class Asset
```

Comment:
I agree. How about `AnalyticsPeriod`? I was considering `Period` (it was called that in my first version, but that also didn't feel right.